Composer/packages/client/src/components/Header.tsx

```diff
@@ -168,6 +187,24 @@ export const Header = () => {
     setAppUpdateShowing(true);
   }, []);
 
+  const hideTeachingBubble = () => {
+    setShowTeachingBubble(false);
+  };
+  const toggleGetStarted = (newvalue) => {
+    hideTeachingBubble();
+    setShowGetStarted(newvalue);
+  };
+
+  // pop out
+  useEffect(() => {
+    if (window.location.hash === '#getstarted') {
+      setshowGetStartedTeachingBubble(true);
+      setShowGetStarted(true);
+    } else {
+      setshowGetStartedTeachingBubble(false);
+    }
+  }, [window.location.hash]);
```

Comment:
I'm pretty sure react can't depend on this value.
You need to do something like this:
```typescript
const useWindowLocationHash = () => {
  const [windowLocationHash, setWindowLocationHash] = React.useState(window.location.hash);
  const listenToPopstate = () => {
    setPath(window.location.hash);
  };
  React.useEffect(() => {
    window.addEventListener("popstate", listenToPopstate);
    return () => {
      window.removeEventListener("popstate", listenToPopstate);
    };
  }, []);
  return windowLocationHash;
};
```

Reply:
then use the hook:
```typescript
const windowLocationHash = useWindowLocationHash();
React.useEffect(() => {
  // your logic
}, [windowLocationHash]);
```

Reply:
It _does_ work as far as I can tell.  Is there some scenario that would cause this not to work that would necesitate doing it in this much more complex way?

Reply:
@benbrown try this
https://codesandbox.io/s/nifty-zhukovsky-tcy1h
Open the console on the bottom right (below page preview)

https://user-images.githubusercontent.com/1483492/113024603-cb9b9080-913b-11eb-9a7c-9a36f622865b.mov

Reply:
I will update to use your suggested solution. Thanks for showing me the details.

I suspect for our current use case directly accessing this value would be ok since it was part of an overall page navigation and the header had a chance to capture it on the first render, but it does not support more sophisticated use  of the hash for navigation which may be the case in the future!

Reply:
Resolved by using `useLocation`